### ISSUE TITLE
NHML overriding GPU temp target

### ISSUE THREAD

**Issue by Renkyz** [nicehash/NiceHashMiner#628]
Haven't been able to find any information to help with my issue.
I use EVGA Precision XOC with 2 EVGA GTX 1070 FTW2 cards.
The problem I'm experiencing is that while mining overnight, at some point the miners start ignoring the temp target (and possibly power target) that I've set in my mining profile inside Precision XOC. This is solved simply by choosing default settings then re applying the profile with my saved settings again.
My concern is that I live in Australia and at this time of year it's getting very hot where I live. I'm rather worried that this is going to happen overnight on a very warm night and my cards will die.
Can anyone help me with figuring out the cause and how to stop it ignoring my EVGA Precision XOC settings.

**Comment by Sav87** [comment]
It looks like the graphics card driver is crashing. If you use overclocking try to reduce it. In extreme cases, use the option "Run script when CUDA GPU is lost"

**Comment by Renkyz** [comment]
I do overclock them but I've tested them very thoroughly and they're stable OC's. I'll reduce them nonetheless and see if the problem continues. Any idea where I could find logs of these crashes? I've just looked in Event Viewer but I'm not sure on what exactly I'm looking for. The only things I can find with "nvidia" in them aren't matching up with dates that my issue has occured.

**Comment by Sav87** [comment]
Event ID 4101 — Display Driver Timeout Detection and Recovery.
windows \ system log

**Comment by Renkyz** [comment]
Just now I looked and my temps were above my temp target once again even though my Precision XOC settings haven't changed. So I just checked and sure enough, almost 30 minutes ago an error with the event ID you gave me is there. Here's what it says:
"Display driver nvlddmkm stopped responding and has successfully recovered."
Dropping my OC hasn't seemed to help at all. Could anything else be causing it? It only seems to happen with NHML (v 1.8.2.0). I never experienced this with NiceHash 2 or Claymore's Dual Ethereum miner (I use to mine ETH+SIA).

**Comment by Sav87** [comment]
this problem occurs when the algorithm is switched. to be precise because of the current implementation of the closing process of the miner.

**Comment by Renkyz** [comment]
Ah ok, I figured it was happening when algorithms were switched but wasn't sure why. Is this the case for all NHML releases? If so I'm assuming it's being worked on for a fix?
Thanks for all the help so far.

**Comment by Sav87** [comment]
it is unlikely to correct it. Although there is a solution, but because of it a log window is propped.

**Comment by DillonN** [comment]
@Sav87 I went over your PR regarding this issue a while ago when you made it, I thought I had left a comment on it but appears it didn't go through or slipped my mind. My main concern with is was it broke the log window and some other minor problems. 

@Renkyz the current switching method has been in place since long before I took over (when NiceHash Miner Legacy began), it is unlikely you'll find a version that uses a different switch. In my testing on multiple NVIDIA systems I could not reproduce the issue with stock settings. Even slight OC I noticed can cause it to happen with increasing frequency however. If users are getting the switching issue on stock settings though we will look into a way of implementing @Sav87's PR that doesn't cause issues with existing features

**Comment by Sav87** [comment]
@DillonN I understood this, so I wrote about the problem with the log window. it can be realized through the option "Log or SoftСlose".

**Comment by Renkyz** [comment]
Thanks for the replies guys, it's helped a lot with me understanding what's going on.
So I've ran the particular rig I've been having the problem with overnight with a reduced OC. When I came back to it everything was fine and I even decided to bump the OC back up to what I found maximally stable for gaming (it's also used as a spare gaming rig for friends/visitors).
If you're curious, the settings I have for both 1070's in Precision XOC are Power Target: 60%, Temp Target: 65C, GPU offset: +100 MHz, Mem offset: +450 MHz.
I also mine on my personal main rig which has an EVGA GTX 1080 FTW and just today I went and purchased the EVGA GTX 1080 Ti which I will eventually be getting in parts to water cool it like my 1080 but for now I have it mining as is, in tandem with my 1080. Both these cards again have gaming stable OC's (still quite high) and I've never experienced the same issue on this PC. I don't know if it's relevant but the only difference between the 2 rigs is the one encountering the problem with the 1070's I use Window 7 Ultimate x64 for mining. Reason being I use to have an old GTX 680 4GB in their because it's hash rate is crap on Windows 10 but have since retired that card.
Sorry for the lengthy post and once again, thank you the help.
